Source/WebCore/Modules/fetch/FetchBodyOwner.cpp

```diff
@@ -138,47 +137,42 @@ void FetchBodyOwner::blob(Ref<DeferredPromise>&& promise)
         return;
     }
     m_isDisturbed = true;
-    m_body->blob(*this, WTFMove(promise), m_contentType);
+    m_body->blob(*this, WTFMove(promise));
 }
 
 void FetchBodyOwner::cloneBody(FetchBodyOwner& owner)
 {
     m_loadingError = owner.m_loadingError;
-
-    m_contentType = owner.m_contentType;
+    m_headers = FetchHeaders::create(owner.m_headers.get());
     if (owner.isBodyNull())
         return;
     m_body = owner.m_body->clone();
 }
 
 ExceptionOr<void> FetchBodyOwner::extractBody(FetchBody::Init&& value)
 {
-    auto result = FetchBody::extract(WTFMove(value), m_contentType);
+    auto currentContentType = contentType();
+    auto wasEmptyContentType = currentContentType.isNull();
+    auto result = FetchBody::extract(WTFMove(value), currentContentType);
+
+    // Initialize the Content-Type header if it was empty.
+    if (wasEmptyContentType)
+        m_headers->fastSet(HTTPHeaderName::ContentType, currentContentType);
```

Comment:
I end up realizing that we should only update the content type to headers when it was empty before running `FetchBody::extract`. If any existing content-type already exists in headers, we should keep it as is.

Reply:
Note that empty and null are not the same thing. We probably want to call this contentTypeIsMissing or some such as that's different from it being the empty string.